### DISCUSSION TITLE
Adding a custom route to the Docusuarus

### DISCUSSION THREAD

**Discussion by flamecopper** [facebook/docusaurus#10280] [General]
Hi there. I would like to add a custom route.
There is a knockdown for when I want to implement a custom button called logout.
When the logout button is pressed, it will actually clear the cookie.
Is there anything I can do to actually implement this?

**Comment by OzakIOne** [comment]
Can you give us the code you tried to implement ?

**Comment by flamecopper** [comment]
Hi there,

On a markdown file, I have created a <form type="submit" action="logout"><input type="type" button="Logout">..., after that I have on the src/component/dbroute.js I have tried to create something like this: "// Placeholder logout route
app.get('/logout', (req, res) => {
    res.clearCookie('user_id');
    res.send('Logged out');
});
"

**Reply by slorber** [reply]
Docusaurus is a static site generator, and will not enable you to implement server-side routes running code in a request/response fashion.

If you want server request features alongside your Docusaurus static site (for example to implement auth), that would be a feature provided by your host, and not Docusaurus.

Here's an example of the usage of Netlify serverless functions for our own needs: https://github.com/facebook/docusaurus/tree/main/admin/new.docusaurus.io

You can find equivalent features on Vercel or Cloudflare platforms
